src/app/Rules/CreateRule.tsx

```diff
@@ -264,12 +265,28 @@ const Comp = () => {
     },
   ];
 
+  const authRetry = React.useCallback(() => {
+    context.target.setAuthRetry();
+  }, [context.target, context.target.setAuthRetry]);
+
   return (
     <BreadcrumbPage pageTitle="Create" breadcrumbs={breadcrumbs}>
       <Grid hasGutter>
         <GridItem xl={7}>
           {errorMessage ? (
-            <ErrorView message={errorMessage} title={'Fail to retrieve event templates'}></ErrorView>
+            <ErrorView
+              message={
+                <>
+                  <Text>{errorMessage}</Text>
+                  {errorMessage === authFailMessage && (
+                    <Button variant="link" onClick={authRetry}>
+                      Retry
+                    </Button>
+                  )}
+                </>
+              }
+              title={'Fail to retrieve event templates'}
```

Comment:
This message snippet is mostly copy-pasted across different components. Can it be extracted as a reuseable component? Maybe extract the whole `ErrorView` into something like `ErrorRetryView`?

Reply:
Right, instead of a passing the entire message component, I refactor it to allow passing `retry` callback. If passed, the `ErrorView ` becomes `ErrorRetryView`.